scripts/generate_docs_sidebar.js

```diff
@@ -28,6 +29,18 @@ const startDirs = [
 const sidebarFile = './docs/_sidebar.md'; // Location to save the generated sidebar
 const excludeDirs = ['node_modules', '.git']; // Directories to exclude from the search
 
+const isGitTrackedMarkdownFile = (() => {
+  let tracked;
+  return (filePath) => {
+    if (tracked === undefined) {
+      tracked = new Set(
+        execSync(`git ls-files '*.md'`).toString().trim().split('\n').filter(Boolean)
```

Comment:
Should this check the status? Like 


```suggestion
        execSync(`git status -s '*.md'`)
```

And if the file is unstaged it will have a space as the first character. In other words:
A file is staged if the first letter is not a space.

That way once we stage the markdown file and trying to commit then it will generate